challenges.yml

```
@@ -1699,3 +1699,448 @@ extending_smart_watcher_2:
           },
           ```
       correct: a
+
+extending_css_transitions:
+  question: |
+    Imagine I have a simple component that can show a message when one of our
+    customers have bought too much pizza.
+
+    ```vue
+    <template>
+        <div :class="{
+                [$style.component]: true,
+                [$style.tall]: showDisclaimer,
+            }"
+        >
+            <button @click="addPizzaToCart" />
+
+            <p v-show="showDisclaimer">
+                <strong>Disclaimer</strong>: Eating too much pizza can be
+                bad for your help!
+            </p>
+        </div>
+    </template>
+
+    <script>
+    import { buyPizza } from '@/services/buy-pizza';
+
+    export default {
+        name: 'PizzaCheckout',
+        data() {
+            return {
+                showDisclaimer: false,
+            };
+        },
+        methods: {
+            async addPizzaToCart() {
+                await buyPizza();
+            }
+        },
+    };
+    </script>
+    ```
+
+    In a Single Page Application, where this component has a fixed height, I want to
+    add some styling so that its height will *transition* smoothly when that message
+    toggles on and off.
+
+    Which `style` tag would be good for the job?
+  explanation: |
+    Don't forget to transition the elements `height` property and provide good values
+    through the different classes!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```vue
+          <style lang="scss" module>
+          .component {
+              height: 200px;
+
+              &.tall {
+                  height: 400px;
+              }
+          }
+          </style>
+          ```
+        b: |
+          ```vue
+          <style lang="scss" module>
+          .component {
+              height: 200px;
+              transition: height;
+
+              &.tall {
+                  height: 400px;
+              }
+          }
+          </style>
+          ```
+        c: |
+          ```vue
+          <style lang="scss" module>
+          .component {
+              height: 200px;
+              transition: height 1s;
+
+              &.tall {
+                  height: 400px;
+              }
+          }
+          </style>
+          ```
+        d: |
+          ```vue
+          <style lang="scss" module>
+          .component {
+              height: 200px;
+              transition: height 1s;
+          }
+          </style>
+          ```
+      correct: c
+
+extending_css_transitions_2:
+  question: |
+    When we transition an elements opacity and try to show or hide this
```

```suggestion
    When we transition an element's opacity and try to show or hide this
```

I'm not sure this question is super clear. I understand what you're asking, but I think the setup is not clear. Maybe you show some simple template + `style` code and ask them what will happen when some variable - e.g. `showDisclaimer` changes?